cfme/tests/test_rest.py

```diff
@@ -69,6 +73,144 @@ def provision_data(
     return result
 
 
+@pytest.fixture(scope='module')
+def service_catalogs_fixture(request, rest_api):
+    name = fauxfactory.gen_alphanumeric()
+    scl = rest_api.collections.service_catalogs.action.add(
+        name=name,
+        description="description_{}".format(name),
+        service_templates=[]
+    )[0]
+    wait_for(
+        lambda: rest_api.collections.service_catalogs.find_by(name=name),
+        num_sec=180,
+        delay=10,
+    )
+
+    service_catalogs_data = [{
+        "name": "name_{}_{}".format(name, index),
+        "description": "description_{}_{}".format(name, index),
+        "service_templates": []
+    } for index in range(1, 5)]
+
+    scls = rest_api.collections.service_catalogs.action.add(*service_catalogs_data)
+
+    def _finished():
+        scl.reload()
+        scl.action.delete()
+        with error.expected("ActiveRecord::RecordNotFound"):
+            scl.action.delete()
+
+        rest_api.collections.service_catalogs.action.delete(*scls)
+        with error.expected("ActiveRecord::RecordNotFound"):
+            rest_api.collections.service_catalogs.action.delete(*scls)
+
+    request.addfinalizer(_finished)
+    return scls
+
+
+@pytest.mark.usefixtures("logged_in")
+@pytest.fixture(scope='module')
+def dialog_fixture():
+    dialog = "dialog_" + fauxfactory.gen_alphanumeric()
+    element_data = dict(
+        ele_label="ele_" + fauxfactory.gen_alphanumeric(),
+        ele_name=fauxfactory.gen_alphanumeric(),
+        ele_desc="my ele desc",
+        choose_type="Text Box",
+        default_text_box="default value"
+    )
+    service_dialog = ServiceDialog(label=dialog, description="my dialog",
+                                   submit=True, cancel=True,
+                                   tab_label="tab_" + fauxfactory.gen_alphanumeric(),
+                                   tab_desc="my tab desc",
+                                   box_label="box_" + fauxfactory.gen_alphanumeric(),
+                                   box_desc="my box desc")
+    service_dialog.create(element_data)
+    return service_dialog
+
+
+@pytest.mark.usefixtures("logged_in")
+@pytest.fixture(scope='module')
+def service_templates_fixture(request, rest_api, dialog_fixture):
+    try:
+        if current_version() <= "5.5":
+            catalog_items = [CatalogItem(item_type="Generic",
```

Comment:
This one deserves a new line for better indent, but I am thinking that maybe even an explicit `for` loop because the parameter list is very long.